portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -57,37 +57,48 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
         int commentsNumber = Integer.parseInt(request.getParameter("number"));
-        
-        List<DataStats> comments = new ArrayList<>();
-        int num = 0;
-        for (Entity entity : results.asIterable()) {
-            long id = entity.getKey().getId();
-            String name = (String) entity.getProperty("name");
-            String comment = (String) entity.getProperty("comment");
-            Date commentTime = (Date) entity.getProperty("commentTime");
-            int upvote = ((Long) entity.getProperty("upvote")).intValue();
-            String imgUrl = (String) entity.getProperty("imgUrl");
-
-            comments.add(new DataStats(name, comment, commentTime, upvote, id, imgUrl));
-            num += 1;
-            if (num == commentsNumber) {
-                break;
+        String json = new String();
+        Gson gson = new Gson(); 
+
+        if (commentsNumber == -1) {
```

Comment:
Sorry I didn't notice this earlier, it looks like you're using the number of comments parameter to distinguish between whether you want to get comments or ratings, which doesn't seem intuitive to me. When the number of comments parameter is set to -1, it's more natural to think that we don't want to limit on the number of comments to return, which means return all comments. For ratings, I think we should use a different endpoint, i.e.: /ratings